samza-core/src/main/java/org/apache/samza/zk/ProcessorData.java

```diff
@@ -51,4 +54,17 @@ public String getHost() {
   public String getProcessorId() {
     return processorId;
   }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(processorId, host);
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (obj == null) return false;
+    if (getClass() != obj.getClass()) return false;
+    final ProcessorData other = (ProcessorData) obj;
+    return processorId.equals(other.processorId) && host.equals(other.host);
```

Comment:
This is just honoring `Object.equals` contract. 

The way in which we want to determine equality(without host affinity) now shouldn't affect ideal implementation. 

For determining duplicate processors, equality check is on processorID alone(Lines 165, 179 in ZkUtils).